### ISSUE TITLE
RuntimeException during installation of composer

### ISSUE THREAD

**Issue by cmodijk** [Herzult/php-ssh#43]
We use deployphp to deploy projects to our servers but sinds a little while composer changed there output from writing lines to writing error's in some cases but the installation is still succesvol. So there is an open issue at deployphp https://github.com/deployphp/deployer/issues/201 library that try's to fix this and they solved it for [PhpSecLib server](https://github.com/deployphp/deployer/commit/f7f51db6f32a5bc54029d738bf79a7da5b5b0d17) implementation but they also use this library to connect to servers. 

In conclusion is it possible to check the output result on exit code like deployphp does for PhpSecLib instead of checking the error output?
* https://github.com/Herzult/php-ssh/blob/master/src/Ssh/Exec.php#L26

More details
* https://github.com/deployphp/deployer/issues/201
* https://github.com/deployphp/deployer/pull/199
* https://github.com/composer/composer/issues/1905#issuecomment-41631659

**Comment by h4cc** [comment]
To avoid confusion here:
You are _using_ php-ssh to connect to a server via SSH and run composer over there.
So there there is no problem _installing_ php-ssh via composer?

And the cause for your ticket is, that composer does some "... older 30 Days build ..." output on STDERR, which is interpreted by this library as a Error and then throws a RuntimeException?

**Comment by cmodijk** [comment]
> You are using php-ssh to connect to a server via SSH and run composer over there.
So there there is no problem installing php-ssh via composer?

Yes

> And the cause for your ticket is, that composer does some "... older 30 Days build ..." output on STDERR, which is interpreted by this library as a Error and then throws a RuntimeException?

Yes :)

**Comment by elfet** [comment]
Is it possible to check return code via php ssh ext?

**Comment by h4cc** [comment]
I checked the SSH2 PHP Documentation and could not find anything describing a way to receive exit codes of executed commands.

I checked:
* http://php.net/manual/de/function.ssh2-exec.php
* http://php.net/manual/de/function.ssh2-shell.php

The only idea i have is something like this:

Appending another command that will output the exit code of the previous command `ssh2_exec($connection, $cmd.' ; echo "PHP_SSH_EXIT_CODE="$?');`.

Some example implementation:
```php
<?php
$username = 'julius';
$password = '';
$server = 'localhost';
$serverPort = 22;

$connection = ssh2_connect($server, $serverPort);
ssh2_auth_password($connection, $username, $password);

executeCommand($connection, 'cd /etc');
executeCommand($connection, 'cd /foo');

function executeCommand($connection, $cmd) {
  echo "\n--- Executing: $cmd \n";
  
  $stream = ssh2_exec($connection, $cmd.' ; echo "PHP_SSH_EXIT_CODE="$?');
  $errorStream = ssh2_fetch_stream($stream, SSH2_STREAM_STDERR);

  stream_set_blocking($errorStream, true);
  stream_set_blocking($stream, true);

  $out = stream_get_contents($stream);
  $err = stream_get_contents($errorStream);

  $exitCode = 0;
  if(preg_match('@PHP_SSH_EXIT_CODE=([0-9]+)@', $out, $matches)) {
    $exitCode = (int)$matches[1];
    $out = str_replace($matches[0], '', $out);
  }

  echo "STDOUT: " . trim($out)."\n";
  echo "STDERR: " . trim($err)."\n";
  echo "ExitCode: ".$exitCode."\n";

  fclose($errorStream);
  fclose($stream);
}
```

**Comment by elfet** [comment]
Looks like only solution.

**Comment by h4cc** [comment]
@Herzult Might that be a solution we could try?

It would need to modify the `Exec::run` signature or introducing a second method.

**Comment by h4cc** [comment]
Solved with: https://github.com/Herzult/php-ssh/issues/43
Contained in release: v1.1.1
